spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -203,4 +207,12 @@ public void testSpecifySnapshotAndTimestamp() {
             .collectAsList();
         });
   }
+
+  @Test
+  public void testBinaryInFilter() {
+    List<Object[]> expected = ImmutableList.of(row(2L, new Byte[]{0x11, 0x11}));
+
+    assertEquals("Should return all expected rows", expected,
+        sql("SELECT id,binary FROM %s where binary > X'11'", binaryTableName));
```

Comment:
Can you add a space between the columns that are being projected?

Reply:
okey, done